codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -692,9 +665,185 @@ void decrementFlowControlWindow(int bytes) {
          * Discards this {@link FlowControlled}, writing an error. If this frame is in the pending queue,
          * the unwritten bytes are removed from this branch of the priority tree.
          */
-        void writeError(FlowControlled frame, Http2Exception cause) {
+        private void writeError(FlowControlled frame, Http2Exception cause) {
             decrementPendingBytes(frame.size());
             frame.error(cause);
         }
     }
+
+    /**
+     * The remote flow control state for a single stream that is not in a state where flow controlled frames cannot
+     * be exchanged.
+     */
+    private static final class ReducedFlowState extends RemoteFlowControlState {
```

Comment:
@nmittler - I could make the `RemoteFlowControlState` implementation provide the functionality in this class.  However it is currently symmetric with the `LocalFlowControlState` and I think a bit more clear in terms of usage to have them separate.